### PR TITLE
feat: added support for multiple security groups.

### PR DESCRIPTION
**Pull request by devnill**
## Overview
Currently, there is an optional parameter `groupName` which is used to ensure that a user belongs to a specific active directory security group. Additionally, when successfully authenticated, the security group is added to the Verdaccio `groups` array for the purpose of role based authentication.

This plugin allows a yaml sequence to be passed in to validate multiple security groups. This allows for more granularity in the authentication process so that the plugin will support multiple levels of access.

### Scoped packages
This will allow users in npmWriters to both publish and unpublish to `@internal-scope`. This is the existing behavior.
```
auth:
  activedirectory:
    url: "ldap://10.0.100.1"
    baseDN: 'dc=sample,dc=local'
    domainSuffix: 'sample.local'
    groupName: 'npmWriters' 

packages:
  '@internal-scope/*':
    access: $all
    publish: npmWriters
    unpublish: npmWriters

  '**':
    access: $all
    proxy: npmjs
```

### Restrictions to unpublish
This will allow users in npmWriters to only publish. Users belonging to npmAdmins may publish and unpublish packages.
```
auth:
  activedirectory:
    url: "ldap://10.0.100.1"
    baseDN: 'dc=sample,dc=local'
    domainSuffix: 'sample.local'
    groupName: 
      - npmWriters
      - npmAdmins

packages:
  '@internal-scope/*':
    access: $all
    publish: npmWriters npmAdmins
    unpublish: npmAdmins

  '**':
    access: $all
    proxy: npmjs
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,3 +20,18 @@ auth:
     domainSuffix: 'sample.local'
     groupName: 'npmWriters' # optional
 ```
```
Comment (nowhammies) [inline README.md:22]: Hey, Thanks for the pull request should we add documentation to the README.rd around the comment you made the the pull request and with the Scoped packages and restrictions to unpublish?

Comment (devnill) [reply README.md:22]: I don't think its needed since it's functionality which is core to Verdaccio itself.

**Comment by nowhammies** [comment]
Ok sounds good. I can merge and post another revision into npm. I'll get
this done by EOD today.

**Comment by nowhammies** [comment]
If the scoped packages stuff needs to be included in the config.yaml we should probably include these examples in the Readme.md. What do you think ?

**Comment by devnill** [comment]
> If the scoped packages stuff needs to be included in the config.yaml we should probably include these examples in the Readme.md. What do you think ?

Scoped packages don't need to be included. I just left that as an example because its a common way to keep internal packages separate from upstream npm repos

**Comment by nowhammies** [comment]
However it is an option so it should be included in an examples section in
the readme.md. I think that makes sense, that way someone doesn't need to
deep dive the verdaccio docs to figure it out. What do you think?

**Comment by devnill** [comment]
Good point. I've added them to the docs with more detailed explanations of how they are used.

**Review by nowhammies — APPROVED** [review]
Looks good I will merge and get this pushed out to npm today.

**Comment by juanpicado** [comment]
@nowhammies @devnill we want to integrate this plugin within our plugin ecosystem to provide a better support, if you will we can https://github.com/verdaccio/monorepo/issues/24 contribute to make this plugin more accessible for more users. Let me know your thoughts,

**Comment by nowhammies** [comment]
Hey Juan i have no issue adding this to you mono repo please feel free to
do so.  Thanks! Sorry for the late reply.

**Comment by juanpicado** [comment]
Thanks ! 🙏

**Pull request merged**